Tools/clinic/clinic.py

```diff
@@ -5294,8 +5294,10 @@ def state_function_docstring(self, line):
         new_docstring += line
         self.function.docstring = new_docstring
 
-    def format_docstring(self):
+    def format_docstring(self) -> str:
         f = self.function
+        assert f is not None
+        assert f.full_name is not None
```

Comment:
I'm not actually sure why we give `full_name` a default value of `None` in `Function.__init__`:

https://github.com/python/cpython/blob/8ebc9fc321ba1eeb3282c2170f351c54956893e6/Tools/clinic/clinic.py#L2528

It seems like we could just make a required argument, then be sure it's always a `str`, and this would make life a lot simpler in several places. But I decided to leave that refactoring for a future PR.

Reply:
Yeah. Let's make it a required arg in a separate PR.